topo_processor/uploader/local.py

```diff
@@ -11,13 +11,16 @@ async def upload_to_local_disk(collection: Collection, target: str):
     if not os.path.isdir(os.path.join(target, collection.title)):
         os.makedirs(os.path.join(target, collection.title))
 
-    for item in collection.items:
+    stac_collection = collection.create_stac()
+    for item in collection.items.values():
         if item.is_valid:
             for asset in item.assets.values():
                 if asset.needs_upload:
                     asset.properties["file:checksum"] = await multihash_as_hex(asset.path)
                     copyfile(asset.path, os.path.join(target, item.parent, f"{item.id}{asset.file_ext}"))
-            await write_stac_metadata(item, os.path.join(target, item.parent, f"{item.id}{item.file_ext}"))
+                    stac_item = item.create_stac()
+                    stac_collection.add_item(stac_item)
+            await write_stac_metadata(stac_item, os.path.join(target, item.parent, f"{item.id}{item.file_ext}"))
         else:
-            get_log().info("Invalid item was not uploaded", source_path=item.source_path, error="TODO")  # TODO
-    await write_stac_metadata(collection, os.path.join(target, collection.title, f"collection{collection.file_ext}"))
+            get_log().warning("Invalid item was not uploaded:", error="; ".join(item.error_msgs), source_path=item.source_path)
```

Comment:
you should be able to log a array, do you need to join the messages with `;`?

Reply:
This is how it looks like if we log the array:
[2021-03-21T08:31:52.412Z] WARN Invalid item was not uploaded: error={ 0=this is an error 1=this is another error } source_path=/home/bcheng/dev/topo-processor/test_data/tiffs/399/CROWN_399_E_51.tiff

This is if we join them:
[2021-03-21T08:32:56.619Z] WARN Invalid item was not uploaded: error=this is an error; this is another error source_path=/home/bcheng/dev/topo-processor/test_data/tiffs/399/CROWN_399_E_51.tiff

Reply:
we will just log the array without joining them